File: backend/gw/src/main/java/com/gi/gateway/dto/request/employee/EmployeeListSearchRequest.java
package com.gi.gateway.dto.request.employee;

import java.util.Date;

import com.gi.gateway.dto.request.common.CommonPaginatorRequest;

import lombok.Getter;

@Getter
public class EmployeeListSearchRequest extends CommonPaginatorRequest {
	private String name;
	private String email;
	private Integer insystemRole;
	private Integer department;
	private String employeeCode;
	private Date dateOfBirth;
	private String generalKeys;
}

File: frontend/GI-HRM-FE/src/app/dto/request/system-staff/employee-list-search-request.ts
import { CommonPaginatorRequest } from "../common-paginator-request";

export class EmployeeListSearchRequest extends CommonPaginatorRequest {
    name: string | undefined;
    email: string | undefined;
    insystemRole: number | undefined;
    department: number | undefined;
    employeeCode: string | undefined;
    entryDate: Date | undefined;
    generalKeys: string | undefined;
}